### DISCUSSION TITLE
nested buttons in headless menu component causes an error

### DISCUSSION THREAD

**Discussion by gone** [alpinejs/alpine#3370] [7. Bugs (archived)]
I have a button component that includes an x-data to ensure it dispatches an event on click. I want to use this component as the button trigger for the headless menu component. If you click the button twice (first time works fine) it will break and stay open. 

See a repro at https://jsfiddle.net/78qu3hmx/

**Comment by imacrayon** [comment]
As a work around you can leave the `x-data` off and `x-on:click` will still work:
```html
<button x-menu:button @click="$dispatch('myEvent')" ...>
```

**Reply by gone** [reply]
Do you mean in general or in this case, since the menu has it's own x-data? 

for this case my workaround was an optional argument to tell the button to leave out the x-data, but it's kind of fiddly since you have to know to do that in this case

**Reply by imacrayon** [reply]
You can leave `x-data` off in all cases because `x-menu:button` already initializes the element as an Alpine component. You don't need any logic to optionally add an `x-data`.

**Reply by gone** [reply]
no, the button component is often used outside of the x-menu case.

**Reply by ekwoka** [reply]
It's not meant to though...

**Comment by xinjie-zhang** [comment]
You could try [Vimesh Headless UI](https://github.com/vimeshjs/vimesh-headless), a more complete Alpine.js headless UI implementation.
